towhee/trainer/trainer.py

```diff
@@ -72,6 +73,31 @@ def reduce_value(value, average=True):
         return value
 
 
+def _construct_from_config(module: Any, config: Union[str, Dict], model=None):
+    """
+    construct from the config, the config can be class name as a `str`, or a dict containing the construct parameters.
+    """
+    instance = None
+    if isinstance(config, str):
+        construct_name = getattr(module, config)
+        if model is not None:
+            instance = construct_name(model.parameters())  # optimizer
+        else:
+            instance = construct_name()  # loss
+    elif isinstance(config, Dict):
+        optimizer_construct_name = config[NAME]
+        construct_name = getattr(module, optimizer_construct_name)
+        kwargs = {}
+        for arg_name in config:
+            if arg_name != NAME:
+                kwargs[arg_name] = config[arg_name]
+        if model is not None:
+            instance = construct_name(model.parameters(), **kwargs)   #optimizer
+        else:
+            instance = construct_name(**kwargs)  # loss
+    return instance
+
+
```

Comment:
It seems optimizer constructor and loss constructors are fused in one functions, will they have conflicts in the future? I suggest it can split in two functions. `_construct_optimizer_from_config` and `_construct_loss_from_config `

Reply:
good suggest, I have solved it.